logger/drain/tcp/drain_test.go

```diff
@@ -0,0 +1,13 @@
+package tcp
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestInvalidDrainUrl(t *testing.T) {
+	_, err := NewDrain("https://wwww.google.com")
+	if err == nil || err.Error() != fmt.Sprintf("Invalid drain url scheme: %s", "https") {
+		t.Error("Did not receive expected error message")
+	}
+}
```

Comment:
Unfortunately this is more than I have time to code up right now (I'm currently packing for a trip to Italy).  I have manually tested both drain implementations and am able to ship logs to Papertrail with no problem.  Can I suggest that you PR this test improvement after this PR is merged?